### PR TITLE
libuv: Upgrade to v1.46.0

### PR DESCRIPTION
**Pull request by no1wudi**
## Summary
libuv: Upgrade to v1.46.0

Rebase the porting code on v1.46.0

## Impact
Minor
## Testing
tested with sim:adb, may not stable enough

### CODE REVIEW

**Review by pkarashchenko — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by pkarashchenko** [comment]
Please fix
```
Undefined symbols for architecture x86_64:
  "_uv__strtok", referenced from:
      _uv__search_path in nuttx.rel
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [nuttx] Error 1
make: *** [nuttx] Error 2
```

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by no1wudi** [comment]
Emm, I don't have such a development environment (x86 base Mac), and can't find where the  `_uv__strtok` defined and called.

**Comment by pkarashchenko** [comment]
let me download your patch and try

**Comment by no1wudi** [comment]
> let me download your patch and try

Thanks

File: system/libuv/Makefile [hunk 1]
```diff
@@ -72,6 +72,7 @@ CSRCS += core.c
 CSRCS += poll.c
 CSRCS += loop.c
 CSRCS += thread.c
+CSRCS += thread-common.c
```
Comment (pkarashchenko) [inline system/libuv/Makefile:75]: Need to add `CSRCS += strtok.c` after `CSRCS += strscpy.c` before `endif`

**Review by pkarashchenko — COMMENTED** [review]
I found how to fix one issue, but some other remain

**Comment by pkarashchenko** [comment]
The other error is related to functionality that is not supported on OSX. So above fix should be enough to pass CI

**Comment by pkarashchenko** [comment]
BTW, strange that the same link error is not reported with Linux `sim:lua` config

**Pull request merged**